src/dcap_provider.cpp

```diff
@@ -898,19 +900,79 @@ static sgx_plat_error_t build_pck_crl_url(
     return SGX_PLAT_ERROR_OK;
 }
 
-static std::string build_tcb_info_url(const std::string& fmspc)
+char* base64_encode(const void* source)
+{
+    // Character set of base64 encoding scheme
+    char char_set[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+    const char* input = static_cast<const char*>(source);
+    int len_str = strlen(input);

```

Comment:
This is not correct. Input is a void pointer and there is an implied assumption in the code that it is string. The param_length parameter should be input to this function and that should be used as length.

Reply:
Please don't assume the input is a string